### DISCUSSION TITLE
Set Default Fields For Relationship

### DISCUSSION THREAD

**Discussion by JaredCostall** [spatie/laravel-query-builder#688] [Q&A]
I'm aware `->allowedFields('user.username')` will allow you to select fields from a relationship in a request.

But is there any way to set the fields that are loaded by default for a relationship? I'm aware `->select('id', 'title')` works for the base table's fields.

A situation occurs when no fields are passed for `allowedFields` so it loads the entire relationship, including fields that I do not want.

I know I could leave these fields out of the User's API Resource, but they are required in some situations, and others not.

**Comment by killjin** [comment]
up
